rule/exported.go

```diff
@@ -330,11 +319,54 @@ func (w *lintExported) Visit(n ast.Node) ast.Visitor {
 		}
 		w.lintTypeDoc(v, doc)
 		w.checkStutter(v.Name, "type")
-		// Don't proceed inside types.
+
+		if w.checkPublicInterface {
+			if iface, ok := v.Type.(*ast.InterfaceType); ok {
+				if ast.IsExported(v.Name.Name) {
+					w.doCheckPublicInterface(v.Name.Name, iface)
+				}
+			}
+		}
+
 		return nil
 	case *ast.ValueSpec:
 		w.lintValueSpecDoc(v, w.lastGen, w.genDeclMissingComments)
 		return nil
 	}
 	return w
 }
+
+func (w *lintExported) doCheckPublicInterface(typeName string, iface *ast.InterfaceType) {
+	for _, m := range iface.Methods.List {
+		// case of ast.Ident and other implicit fields
+		if len(m.Names) == 0 {
+			continue
+		}
+		if ast.IsExported(m.Names[0].Name) {
+			w.lintInterfaceMethod(typeName, m)
+		}
+	}
+}
+
+func (w *lintExported) lintInterfaceMethod(typeName string, m *ast.Field) {
+	name := m.Names[0].Name
+	if m.Doc == nil {
+		w.onFailure(lint.Failure{
+			Node:       m,
+			Confidence: 1,
+			Category:   "comments",
+			Failure:    fmt.Sprintf("public interface method %s.%s should be commented", typeName, name),
+		})
+		return
+	}
+	s := normalizeText(m.Doc.Text())
+	prefix := m.Names[0].Name + " "
```

Comment:
This line needs a comment to explain why you are doing this.

I get it, but it would be better.

But this pattern is common in the current code, you are only reusing it.

So maybe a func hasComment could help.

But it's somehow out of the scope of this PR

Reply:
yeah, i have not apply much refactoring because of "tradition"